server/api/users/users.test.ts

```diff
@@ -37,19 +37,66 @@ describe('auth router', () => {
   });
 
   describe('GET /', () => {
-    it('expects empty array if no users exist', async () => {
+    it('expects empty array if no users exist on the homepage', async () => {
       const res = await server.exec.get('/api/users/');
-      expect(res.body).toStrictEqual([]);
+      expect(res.body.users).toStrictEqual([]);
     });
 
-    it('returns all users', async () => {
-      const userOne = await createUser();
-      const userTwo = await createUser();
+    it('expect invalid page request to return invalid page', async () => {
+      const res = await server.exec.get('/api/users/?page=-1');
+      expect(res.status).toBe(400);
+      expect(res.body.error).toEqual('Invalid page number');
+    });
+
+    it('expect page request beyond page limit to be out of range', async () => {
+      const res = await server.exec.get('/api/users/?page=100');
+      expect(res.status).toBe(400);
+      expect(res.body.error).toEqual('Page out of range');
+    });
+
+    it('returns first 20 users when it hits the homepage', async () => {
+
+      const users: User[] = [];
+
+      for(let i = 0; i < 25; i++){
+        const user = await createUser();
+        users.push(user);
```

Comment:
I could not develop a way to have users stay between this test and the test that starts on line 79. It looks like @timmyichen is good at foreshadowing potential issues with the afterEach deleting users.

Open to suggestions, or we can refactor after we solve the testing/db issue.

Reply:
i think it's fine to just do what you're doing (just parallelize it a bit), as long as you're not seeing a big perf hit

Reply:
I wouldn't worry about it... you obviously don't want to have to seed a lot of stuff into the database on each test, but it won't hurt to do it a few times.